### ISSUE TITLE
"null" as Parameter Name Throws Incorrect Error

### ISSUE THREAD

**Issue by AndyShortt** [aws-cloudformation/cfn-lint#3067]
### CloudFormation Lint Version

0.85.2_1

### What operating system are you using?

Mac

### Describe the bug

Setting a Parameter name to "null" produces incorrect or misleading errors:
"E0002 Unknown exception while processing rule E2011: object of type 'NoneType' has no len()"
"E0002 Unknown exception while processing rule E2003: expected string or buffer"
(further, the document line reference is 1:1; which is not correct)

The Cloudformation service itself returns the following more specific error message:
"Template format error: [/Parameters] encountered malformed key"

### Expected behavior

cfn-lint returns correct line number and a more accurate error message, such as:
"E000X Parameters encountered malformed key"

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09

Parameters:
  null:
    Description: anything
    Type: String

#...
```

**Comment by kddejong** [comment]
This fix will raise an error during a parse if a key is null.  null in json will fail during parse.
